### PR TITLE
Alert Filters Minor Cleanup and Fix

### PR DESCRIPTION
**Pull request by kingthorin**
- DialogAddAlertFilter - clearFields(), now resets all relevant field,
addressing an issue reported in IRC. Removed some unused commented code.
- AlertFilter - Various JavaDoc fixes (remove references to "user")
- AlertFilterAPI - Various JavaDoc fixes (remove references to "user"),
remove unused import.
- AlertFilterTableModel - Various JavaDoc fixes (remove references to
"user").
- ExtensionAlertFilters - Various JavaDoc fixes (remove references to
"user").
- ContextAlertFilterManager - Minor JavaDoc updates.
- DialogModifyAlertFilter - Remove placeholder JavaDoc constructs.

- ZapAddOn.xml - Updated to v4 and new change comment added.

### CODE REVIEW

File: src/org/zaproxy/zap/extension/alertFilters/DialogAddAlertFilter.java [hunk 1]
```diff
@@ -131,6 +128,11 @@ protected void performAction() {
 	@Override
 	protected void clearFields() {
 		this.enabledCheckBox.setSelected(true);
+		this.alertCombo.setSelectedIndex(0);
+		this.newLevelCombo.setSelectedIndex(0);
+		this.urlTextField.setText("");
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/alertFilters/DialogAddAlertFilter.java]: Should also `discardAllEdits()`.

File: src/org/zaproxy/zap/extension/alertFilters/DialogAddAlertFilter.java [hunk 2]
```diff
@@ -131,6 +128,11 @@ protected void performAction() {
...
+		this.regexCheckBox.setSelected(false);
+		this.paramTextField.setText("");
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/alertFilters/DialogAddAlertFilter.java]: Same here, should discard the edits.

File: src/org/zaproxy/zap/extension/alertFilters/ExtensionAlertFilters.java [hunk 3]
```diff
@@ -292,8 +289,8 @@ public AbstractContextPropertiesPanel getContextPanel(Context ctx) {
 	/**
 	 * Gets the context panel for a given context.
 	 * 
-	 * @param contextId the context id
-	 * @return the context panel
+	 * @param contextId the context id the panel should relates to
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/alertFilters/ExtensionAlertFilters.java]: Is the addition right? (i.e. `should relates`)

Comment (kingthorin) [inline src/org/zaproxy/zap/extension/alertFilters/ExtensionAlertFilters.java]: Oops

**Comment by thc202** [comment]
Looks good to me, thanks!

**Pull request merged**